lib/diaspora_federation/entities/account_migration.rb

```diff
@@ -19,20 +21,39 @@ class AccountMigration < Entity
       entity :profile, Entities::Profile
 
       # @!attribute [r] signature
-      #   Signature that validates original and target diaspora* IDs with the new key of person
+      #   Signature that validates original and target diaspora* IDs with the private key of the second identity, other
+      #   than the entity author. So if the author is the old identity then this signature is made with the new identity
+      #   key, and vice versa.
       #   @return [String] signature
       property :signature, :string, default: nil
 
-      # @return [String] string representation of this object
-      def to_s
-        "AccountMigration:#{author}:#{profile.author}"
+      # @!attribute [r] old_identity
+      #   Optional attribute which keeps old diaspora* ID. Must be present when author attribute contains new diaspora*
+      #   ID.
+      #   @return [String] old identity
+      property :old_identity, :string, optional: true
+
+      # Returns diaspora* ID of the old person identity.
+      # @return [String] diaspora* ID of the old person identity
+      def old_identity
+        return old_identity_prop if author_is_new_id?
```

Comment:
👍 (I would have missed that, but `@old identity || author` could be dangerous, when author + new indentity  is used for the signatures (because they aren't equal), but `old_identity` returns  something different)